crates/router/src/bin/router.rs

```diff
@@ -39,13 +39,11 @@ async fn main() -> ApplicationResult<()> {
     logger::info!("Application started [{:?}] [{:?}]", conf.server, conf.log);
 
     #[allow(clippy::expect_used)]
-    let (server, mut state) = router::start_server(conf)
+    let (server, _) = router::start_server(conf)
```

Comment:
Sorry for interrupting but I have a small question, 

In `start_server` function the returned value is a clone of the `state`. 

So here if a `state` clone is dropped, and that triggers the redis_connection pool drop function. 

Does that mean anywhere in the code base if a `state` clone is dropped then redis connection is dropped(closed) ?

Reply:
Since state has `Arc<RedisConnectionPool>` dropping would decrement the reference counter

Reply:
Got it. Thanks.